### ISSUE TITLE
[bug] Responsive props (e.g `$gtSm`) don't work in native.

### ISSUE THREAD

**Issue by jonsherrard** [tamagui/tamagui#420]
**Describe the bug**

On expo/in-native responsive props on Tamagui elements do not work at all

Repro here: https://github.com/jonsherrard/responsive-prop-repro

Example component: https://github.com/jonsherrard/responsive-prop-repro/blob/main/packages/ui/src/MyComponent.tsx

**To Reproduce**

`npx create-tamagui-app@latest`

Add a prop to your component, compare in Next.js and Expo

Try to use the object syntax with a `$xs` or `$gtLg` style prop on any tamagui element.

Repro: https://github.com/jonsherrard/responsive-prop-repro

## Video
https://user-images.githubusercontent.com/631670/206288311-596b99f6-aebc-4169-b466-b45d0840bc0d.mp4

**System Info (please complete the following information):** 

Output of `npx envinfo --system --npmPackages --binaries --browsers`

```sh
  System:
    OS: macOS 12.6
    CPU: (16) x64 Intel(R) Xeon(R) W-2140B CPU @ 3.20GHz
    Memory: 425.63 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.17.0 - ~/.nvm/versions/node/v16.17.0/bin/node
    Yarn: 3.2.3 - ~/.yarn/bin/yarn
    npm: 2.15.12 - ~/node_modules/.bin/npm
    Watchman: 2022.07.04.00 - /usr/local/bin/watchman
  Browsers:
    Brave Browser: 107.1.45.123
    Chrome: 108.0.5359.94
    Firefox: 107.0
    Safari: 16.0
  npmPackages:
    @babel/runtime: ^7.18.9 => 7.20.6 
    @manypkg/cli: ^0.19.1 => 0.19.2 
    @types/react: ^18.0.15 => 17.0.21 
    @types/react-native: ^0.69.6 => 0.69.15 
    eslint: ^8.21.0 => 8.28.0 
    prettier: ^2.7.1 => 2.8.0 
    turbo: ^1.4.2 => 1.6.3 
    typescript: ^4.7.4 => 4.9.3 
```

**Comment by natew** [comment]
Just tested in the repo and it's working, maybe in the starter itself 🤔

**Comment by jonsherrard** [comment]
Yeah I found that too. I tried to setup a standalone expo app with Tamagui, but couldn’t get that working due to some issue with the portal dependency.

Any pointers on what I should look at? Struggled for a couple of weeks now!

Do you experience the issue if you create a starter?

**Comment by jonsherrard** [comment]
I did a bit of cleaning up https://github.com/jonsherrard/tamagui-test/pull/1 to compare code between the `apps/starter` and `starter/expo-solito` directories.

The main difference I could see as a culprit would be the switch from using a "built" version of the UI to loading the source directly. 

I will continue my investigations...

**Comment by jonsherrard** [comment]
> Just tested in the repo and it's working, maybe in the starter itself 🤔

Which part of the repo did you test in out of interest @natew?

**Comment by natew** [comment]
Try out rc1, has a variety of improvements for media in general.

**Comment by natew** [comment]
I did replicate on starter, on a plane right now but in my manual test the update fixed it. I'll test from scratch later tonight.

**Comment by bureyburey** [comment]
Can confirm rc1 fixed this for me 👑

**Comment by natew** [comment]
Dope. There was one more issue rc1.1 fixes

**Comment by jonsherrard** [comment]
Oh man... tearing my hair out. I swear I'm not at idiot 😂 

I can see all the changes Nate (thank you), and obviously @bureyburey has it working...

But when I `create-tamagui-app` with `rc-1.1`, none of the changes I make using `$xs` or `$gtSm` props are working on Expo.

<img width="1355" alt="image" src="https://user-images.githubusercontent.com/631670/206699209-6ce24792-01e0-4611-ad6d-a63fd7c60e4f.png">

This is happening on my iMac Pro and my Macbook Pro (both 2019), and whether I run the code on a device (iPhone 14 Pro) or the Simulator.

Maybe this is something to do with Expo!? 

I'm going to go and delete Expo from my machine and phone and see if I can make any progress there - thought I'd document my plight incase anyone has the same issues.

**Comment by jonsherrard** [comment]
Petition to re-open this @natew 

I saw in the sandbox your test was this:


```tsx
  <XStack $xs={{ fd: 'column' }} space>
        <MyComponent w={50} h={50} />
        <MyComponent w={50} h={50} blue />
        <MyComponent w={50} h={50} />
      </XStack>
```

BUT, this isn't actually applying the `flex-direction: column` property on native, it's _breaking_ the underlying property on `XStack`, which then defaults back to the default for `Stack` which is `flex-direction: column` - which gives the expected behaviour - but it's a false positive I think!

You can test this by doing:

```tsx
 <MyComponent w={50} h={50} $xs={{bc: 'green'}} />
 ```
 
 ^ the box will go blank
 
 OR:
 
```tsx
// background-color: green will NOT be applied 
  <XStack $xs={{ bc: 'green', f }} space>
    <MyComponent w={50} h={50} />
    <MyComponent w={50} h={50} blue />
     <MyComponent w={50} h={50} />
 </XStack>
 ```
 
 The box background color will disappear, as the underlying property is now "broken"

**Comment by natew** [comment]
All good. I’ll check out further must be something with the starter specifically.

**Comment by peterjskaltsis** [comment]
I seem to be getting this as well after the `createMedia` release 👀

**Comment by natew** [comment]
Fix for this is out in `rc1.2` sorry for the troubles, the naming inside that getSplitStyle function is prone to errors as we have `key`, `keyInit`, `subKey` and `mediaKey`, I was mixing up key and subKey :/

Tests are the answer to all problems, will be adding more integration tests, we should get a few visual tests, and a lot of unit tests on getSplitStyle as it's an especially sensitive area.

Actually those getSplitStyle unit tests should be fun to write and super useful and very easy to contribute if anyone wants to help getting a few in. See `ThemeManager.test.tsx` and just run `yarn test` in core.

**Comment by jonsherrard** [comment]
YES!

@natew you are a legend thank you. I've added 

```
media: createMedia({
    sx: { minWidth: 100 },
```
    
 To my config and now i can use `$sx` to stsyle everything a-la `theme-ui`, without using variants, and I've got all the magic of `gtSm` and `gtLg` for styling larger screens on web.
 
 It's BEAUTIFUL
 
 If you have any ideas for visual/native testing fraemworks you want to use, I'd be happy to try and contribute some tests.
 
 Thanks again!
